src/pal/src/loader/module.cpp

```diff
@@ -1406,6 +1407,37 @@ static LPWSTR LOADGetModuleFileName(MODSTRUCT *module)
     return module->lib_name;
 }
 
+static bool ShouldRedirectToCurrentLibrary(LPCSTR libraryNameOrPath)
+{
+    if (!g_running_in_exe)
+        return false;
+        
+    if (strcmp(libraryNameOrPath, "[self]") == 0)
+        return true;
+
+    const char *toRedirect[] = {
+        "System.Native.so",
+        "System.Globalization.Native.so",
+        "System.IO.Compression.Native.so",
+        //"System.IO.Ports.Native.so",
+        "System.Net.Http.Native.so",
```

Comment:
We should not be including `System.Net.Http.Native.so` here and statically linking it. It is for legacy Curl HTTP networking stack. Static linking to it has problems - https://github.com/dotnet/corert/issues/7000 has details.

It is fine to not support the Curl HTTP networking stack for single file, I think.

Reply:
> It is fine to not support the Curl HTTP networking stack for single file, I think.

I would like to remove the curl implementation entirely post-3.0.

Reply:
Removed